db/bigtable_eth1.go

```diff
@@ -1609,6 +1609,74 @@ func (bigtable *Bigtable) TransformUncle(block *types.Eth1Block, cache *freecach
 	return bulkData, bulkMetadataUpdates, nil
 }
 
+// TransformWithdrawals accepts an eth1 block and creates bigtable mutations.
+// It transforms the withdrawals contained within a block, extracts the necessary information to create a view and writes that information to bigtable
+// It writes uncles to table data:
+// Row:    <chainID>:W:<reversePaddedNumber>:<withdrawalIndex>
+// Family: f
+// Column: data
+// Cell:   Proto<Eth1WithdrawalIndexed>
+// Example scan: "1:W:" returns withdrawals in desc order
+// Example scan: "1:W:984886725" returns mainnet withdrawals included after block 15113275 (1000000000 - 984886725)
+//
+// It indexes withdrawals by:
+// Row:    <chainID>:I:W:<Address>:TIME:<reversePaddedBigtableTimestamp>
+// Family: f
+// Column: <chainID>:W:<reversePaddedNumber>
+// Cell:   nil
+// Example lookup: "1:I:W:ea674fdde714fd979de3edf0f56aa9716b898ec8:TIME:" returns withdrawals received by ethermine in desc order
+func (bigtable *Bigtable) TransformWithdrawals(block *types.Eth1Block, cache *freecache.Cache) (bulkData *types.BulkMutations, bulkMetadataUpdates *types.BulkMutations, err error) {
+	bulkData = &types.BulkMutations{}
+	bulkMetadataUpdates = &types.BulkMutations{}
+
+	for i, withdrawal := range block.Withdrawals {
+		if i > 16 {
+			return nil, nil, fmt.Errorf("unexpected number of withdrawals in block expected at most 16 but got: %v", i)
+		}
+		iReversed := reversePaddedIndex(i, 10)
+
+		withdrawalIndexed := types.Eth1WithdrawalIndexed{
+			BlockNumber:    block.Number,
+			Index:          withdrawal.Index,
+			ValidatorIndex: withdrawal.ValidatorIndex,
+			Address:        withdrawal.Address,
+			Amount:         withdrawal.Amount,
+			Time:           block.Time,
+		}
+
+		bigtable.markBalanceUpdate(withdrawal.Address, []byte{0x0}, bulkMetadataUpdates, cache)
+
+		// store uncles in with the key <chainid>:W:<reversePaddedBlockNumber>:<reversePaddedWithdrawalIndex>
```

Comment:
store the withdrawals*

Reply:
ah, you are using reversed withdrawalindex, update comment above then por favor

Reply:
done